tests/test_utils.py

```diff
@@ -147,10 +145,58 @@ def test_file_path_validator_with_invalid_paths(actual_path):
 def test_file_path_validator_with_valid_paths(actual_path):
     from prompt_toolkit.document import Document
 
-    validator = rasa.utils.io.questionary_file_path_validator(
-        [".yml", ".yaml"], "error message"
-    )()
+    validator = rasa.utils.io.file_type_validator([".yml", ".yaml"], "error message")
 
     document = Document(actual_path)
     # If the path is valid there shouldn't be an exception
     assert validator.validate(document) is None
+
+
+@pytest.mark.parametrize("input", ["", "   ", "\t", "\n"])
+def test_non_empty_text_validator_with_empty_input(input):
+    from prompt_toolkit.validation import ValidationError
+    from prompt_toolkit.document import Document
+
+    test_error_message = "enter something"
+
+    validator = rasa.utils.io.not_empty_validator(test_error_message)
+
+    document = Document(input)
+    with pytest.raises(ValidationError) as e:
+        validator.validate(document)
+
+    assert e.value.message == test_error_message
+
+
+@pytest.mark.parametrize("input", ["utter_greet", "greet", "Hi there!"])
+def test_non_empty_text_validator_with_valid_input(input):
+    from prompt_toolkit.document import Document
+
+    validator = rasa.utils.io.not_empty_validator("error message")
+
+    document = Document(input)
+    # If there is input there shouldn't be an exception
+    assert validator.validate(document) is None
+
+
+def test_create_validator_from_callable():
+    from prompt_toolkit.validation import Validator, ValidationError
+    from prompt_toolkit.document import Document
+
+    def is_valid(input) -> None:
```

Comment:
you could also write that as `lambda user_input: user_input == "this passes"`

Reply:
or just `return user_input == "this passes"` I suppose 😅